### PR TITLE
fix : fixed broken unit tests for `jkube-kit/common` on windows

### LINKED ISSUE #2563: Fix `jkube-kit/common` module's unit tests on Windows
### Component

JKube Kit

### Description
Related to https://github.com/eclipse/jkube/issues/1338

:warning: A windows machine is required to reproduce and fix problems in this issue.

When running `mvn clean install` on Windows, I see the following tests failing in `jkube-kit/common` module:
- [ExternalCommandTest](https://github.com/eclipse/jkube/blob/master/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ExternalCommandTest.java)
- [ArchiveDecompressorTest](https://github.com/eclipse/jkube/blob/master/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/archive/ArchiveDecompressorTest.java)
- [IoUtilTest](https://github.com/eclipse/jkube/blob/master/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/IoUtilTest.java)
- [SerializationTest](https://github.com/eclipse/jkube/blob/master/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SerializationTest.java)

#### ExternalCommandTest
It tests a utility class of executing a command. We're trying to execute unix commands on Windows, which seems to cause tests to fail. We need to add `@DisabledOnOs(OS.WINDOWS)` annotation for tests executing unix commands.

#### ArchiveDecompressorTest
We seem to be passing unix file separator in expectations, this would never match a path on windows:
https://github.com/eclipse/jkube/blob/31ef8beec2b642be4a8516e425ecdad40ddfb7a8/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/archive/ArchiveDecompressorTest.java#L69
We can rely on [Apache's separatorsToSystem](https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FilenameUtils.html#separatorsToSystem(java.lang.String)) utility method to convert it to windows file separator
```
.containsExactlyInAnyOrder(parentDir, separatorsToSystem(artifactParentDir,) separatorsToSystem(artifact)); 
```

#### IoUtilTest
This is similar to issue faced above, wrapping second argument in separatorsToSystem method should fix this issue:
https://github.com/eclipse/jkube/blob/31ef8beec2b642be4a8516e425ecdad40ddfb7a8/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/IoUtilTest.java#L143

#### SerializationTest
Here we're trying to compare file contents but we have hardcoded UNIX line separator character `\n` in string literal used in expectations:
https://github.com/eclipse/jkube/blob/31ef8beec2b642be4a8516e425ecdad40ddfb7a8/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SerializationTest.java#L160-L164

We can rely on `String.format(` and replace `\n` by `%n` to make it work on windows:
```java
      .isEqualTo(String.format("{%n" +
        "  \"apiVersion\" : \"v1\",%n" +
        "  \"kind\" : \"ConfigMap\",%n" +
        "  \"metadata\" : {%n" +
        "    \"name\" : \"test\"%n" +
        "  },%n" +
        "  \"data\" : {%n" +
        "    \"key\" : \"value\"%n" +
        "  }%n" +
        "}"));
```

### Expected Behavior
All tests pass in `jkube-kit/common` module

### Acceptance Criteria
- [ ] `jkube-kit/common` module's build passes on windows

### Before you start :red_circle:

:point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down::point_down::point_down::point_down::point_down::point_down::point_down:
Make sure you read the [contributing guide](https://www.eclipse.org/jkube/contributing) first. Pay special attention to the ECA agreement section and the requirement to sign-off your commit.

#### Kubernetes

If you don't have a real Kubernetes cluster available (most probably), you can use Minikube or Kind to test with a local cluster.

- Minikube [get started guide](https://minikube.sigs.k8s.io/docs/start/)
- Kind (Requires Docker) [quick start](https://kind.sigs.k8s.io/docs/user/quick-start/)

#### OpenShift

If you don't have a real OpenShift cluster available (most probably), you can use [Red Hat's developer Sandbox for Red Hat OpenShift](https://developers.redhat.com/developer-sandbox). The only requirement is to have a Red Hat account.

Once you have your Sandbox environment, you'll need to download the oc tool from the cluster console.
(Press the `?` icon and from the context menu select `Command line tools`, you'll be redirected to https://$subdomain.openshiftapps.com/command-lines-tools where you'll be able to download the CLI for your platform)

### PR DESCRIPTION
**Pull request by l3002**
## Description

Fixes #2563 : Fixes failing unit tests for windows in ```jkube-kit/common``` package.

All the units tests are passing while running ```mvn clean install``` on Windows now.

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/jkubeio/ci)

Started new GH workflow run for https://github.com/eclipse/jkube/pull/2574 (_2024-01-22T23:49:25Z_)

:gear: [JKube E2E Tests (7618869592)](https://github.com/jkubeio/ci/actions/runs/7618869592) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#2574)](https://github.com/jkubeio/ci/actions/runs/7618869592/job/20750868813)

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/archive/ArchiveDecompressorTest.java [hunk 1]
```diff
@@ -66,7 +67,7 @@ void extractArchive_whenArchiveWithNestedDir_thenExtractToSpecifiedDir(String fi
     FileAssertions.assertThat(tempDir)
         .exists()
         .fileTree()
-        .containsExactlyInAnyOrder(parentDir, artifactParentDir, artifact);
+            .containsExactlyInAnyOrder(parentDir, separatorsToSystem(artifactParentDir), separatorsToSystem(artifact));
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/archive/ArchiveDecompressorTest.java:70]: indentation slightly off here

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/IoUtilTest.java [hunk 2]
```diff
@@ -140,7 +141,7 @@ void downloadArchive_whenUnixArtifactProvided_thenDownloadAndExtract() throws IO
             FileAssertions.assertThat(temporaryFolder)
                 .exists()
                 .fileTree()
-                .containsExactlyInAnyOrder("linux-amd64", "linux-amd64/foo");
+                    .containsExactlyInAnyOrder("linux-amd64", separatorsToSystem("linux-amd64/foo"));
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/IoUtilTest.java:144]: indentation slightly off here

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SerializationTest.java [hunk 3]
```diff
@@ -157,16 +157,16 @@ void asJson_withConfigMap_returnsJsonString() {
...
+            .isEqualTo(String.format("{%n" +
+                    "  \"apiVersion\" : \"v1\",%n" +
+                    "  \"kind\" : \"ConfigMap\",%n" +
+                    "  \"metadata\" : {%n" +
+                    "    \"name\" : \"test\"%n" +
+                    "  },%n" +
+                    "  \"data\" : {%n" +
+                    "    \"key\" : \"value\"%n" +
+                    "  }%n" +
+                    "}"));
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SerializationTest.java:169]: Could you please revert formatting-related changes?

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SerializationTest.java [hunk 4]
```diff
@@ -182,16 +182,16 @@ void saveJson_withConfigMap_savesFile(@TempDir Path targetDir) throws IOExceptio
...
```
[the remark this answers is not in the document]
Comment (l3002) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SerializationTest.java:194]: Sure, just to clarify you me the indentation related changes right?

[the remark this answers is not in the document]
Comment (rohanKanojia) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SerializationTest.java:194]: yes

**Comment by rohanKanojia** [comment]
@l3002 : Could you please rebase your branch against latest updated master?

**Review by manusa — APPROVED** [review]

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**